### DISCUSSION TITLE
Priority rule.

### DISCUSSION THREAD

**Discussion by BobSquarePants** [evilsocket/opensnitch#596] [Q&A] [answered]
On the tooltip of the Priority rule we can read
```
...no others rules will be checked after this one
```
![oh](https://user-images.githubusercontent.com/23012674/150533755-5aaba0c3-2738-4c72-80ad-f2b96035ed99.png)

So If I understand well, if we create a rule like this one
![115916860-addcf500-a475-11eb-86f4-af2c645aa2ba](https://user-images.githubusercontent.com/23012674/150533911-8f9cff69-5d83-48c1-b8df-dd06605da90d.png)
seen at: https://github.com/evilsocket/opensnitch/wiki/block-lists#how-to-add-a-global-rule-to-block-ads-or-trackers

all the following rules will not be process anymore ? is that correct ?

**Comment by gustavo-iniguez-goya** [comment, marked as answer]
That's correct @BobSquarePants :+1: 

If it's not clear or can be explained better, please, suggest a new sentence!

**Reply by BobSquarePants** [reply]
No, I think it's clear.

But is it required in order to: " have a global rule to block ads or trackers " ?
https://github.com/evilsocket/opensnitch/wiki/block-lists#how-to-add-a-global-rule-to-block-ads-or-trackers

If you don't tick **Priority** , will it work ?

if not why ?

**Reply by BobSquarePants** [reply]
No, actually is not that clear. is "No other rules will be checked after this one" mean
always (so meaning as soon as that Priority rule exist no other rule will be ever checked)
or only if that rule trigger ?

**Reply by gustavo-iniguez-goya** [reply]
ok, let me explain how it works:

- When a Deny rules matches against a connection -> the connection is denied right away, no others rules are checked.
- If the connection matches against an Allow rule, it'll allowed. In this case, all the rules are checked.
- If the connection doesn't match with any of the rules, ask the user to allow/deny it.

Rules are checked in alphabetical order.

If you want to prioritize a Deny rule in order to improve performance, name that rule `000-deny-rule`, or `AAA - whatever`
Otherwise, if the Deny rule is at position 7th, OpenSnitch has to check 6 rules before deny the connection. 

Also if you want to prioritize an Allow rule, you must mark `[x] Priority rule` + name the rule accordingly. The priority flag makes OpenSnitch stop checking the rest of the rules, applying the action of that rule if it matches.
